code/messaging/src/phone/java/com/adobe/marketing/mobile/messaging/MessagingUtils.java

```diff
@@ -0,0 +1,162 @@
+     *
+     * @param list A {@link List<T>} to be converted to a mutable list
+     * @return the mutable {@link List<T>} list
+     */
+    private static <T> List<T> createMutableList(final List<T> list) {
+        return new ArrayList<>(list);
+    }
+
+    /**
+     * Updates the provided {@code Map<Surface, List<Proposition>>} with the provided {@code Surface} and {@code List<Proposition>} objects.
+     *
+     * @param surface           A {@link Surface} key used to update a {@link List<Proposition>} value in the provided {@link Map<Surface, List<Proposition>>}
+     * @param propositionsToAdd A {@link List<Proposition>} list to add in the provided {@code Map<Surface, List<Proposition>>}
+     * @param mapToUpdate       The {@code Map<Surface, List<Proposition>>} to be updated with the provided {@code Surface} and {@code List<Proposition>} objects
+     * @return the updated {@link Map<Surface, List<Proposition>>} map
+     */
+    @RestrictTo(RestrictTo.Scope.SUBCLASSES)
+    public static Map<Surface, List<Proposition>> updatePropositionMapForSurface(final Surface surface, final List<Proposition> propositionsToAdd, Map<Surface, List<Proposition>> mapToUpdate) {
+        if (isNullOrEmpty(propositionsToAdd)) {
+            return mapToUpdate;
+        }
+        final Map<Surface, List<Proposition>> updatedMap = new HashMap<>(mapToUpdate);
+        final List<Proposition> list = updatedMap.get(surface) != null ? updatedMap.get(surface) : createMutableList(propositionsToAdd);
+        if (updatedMap.get(surface) != null) {
+            list.addAll(propositionsToAdd);
+        }
+        updatedMap.put(surface, list);
+        return updatedMap;
+    }
+
+    /**
+     * Updates the provided {@code Map<Surface, List<Proposition>>} map with the provided {@code Surface} and {@code Proposition} objects.
+     *
+     * @param surface     A {@link Surface} key used to update a {@link List<Proposition>} value in the provided {@link Map<Surface, List<Proposition>>}
+     * @param proposition A {@link Proposition} object to add in the provided {@code Map<Surface, List<Proposition>>}
+     * @param mapToUpdate The {@code Map<Surface, List<Proposition>>} to be updated with the provided {@code Surface} and {@code List<Proposition>} objects
+     * @return the updated {@link Map<Surface, List<Proposition>>} map
+     */
+    @RestrictTo(RestrictTo.Scope.SUBCLASSES)
+    public static Map<Surface, List<Proposition>> updatePropositionMapForSurface(final Surface surface, final Proposition proposition, Map<Surface, List<Proposition>> mapToUpdate) {
+        if (proposition == null) {
+            return mapToUpdate;
+        }
+        final Map<Surface, List<Proposition>> updatedMap = new HashMap<>(mapToUpdate);
+        final List<Proposition> list = updatedMap.get(surface) != null ? updatedMap.get(surface) : createMutableList(proposition);
+        if (updatedMap.get(surface) != null) {
```

Comment:
nit: This is not efficient as you are making three `get` calls on the Map when you can just do with a single call.

Reply:
fixed this in the internal and public messaging utils